torch/csrc/tensor/python_tensor.cpp

```diff
@@ -405,7 +405,11 @@ void py_set_default_dtype(PyObject* obj) {
       [backend, scalar_type](PyTensorType *x) {
         return x->get_backend() == backend && x->get_scalar_type() == scalar_type;
       });
-    set_default_tensor_type(*it);
+    if (it != tensor_types.end()) {
```

Comment:
I think this is unnecessary because `THPDtype_Check` takes care of it

Reply:
wait, but that's exactly the reason it used to segfault before. end was dereferenced, so a garbage value was passed to set_default_tensor_type, which was dereferenced again causing a segfault. am i missing something?

Reply:
i mean, in general you're supposed to check return values of things, so i'm in favor of keeping this regardless

Reply:
yup so I'd just add `TORCH_INTERNAL_ASSERT(it != tensor_types.end())` because we expect to find a non null value for it, and if that's not the case then there's something wrong in the internal function behavior, and not a user code issue.

Reply:
yep, will do

Reply:
done, please take a look